src/v/cloud_storage/partition_manifest.cc

```diff
@@ -1357,6 +1357,37 @@ void partition_manifest::serialize(std::ostream& out) const {
     serialize_end(c);
 }
 
+ss::future<>
+partition_manifest::serialize(ss::output_stream<char>& output) const {
+    auto iso = _insync_offset;
+    iobuf serialized;
+    iobuf_ostreambuf obuf(serialized);
+    std::ostream os(&obuf);
+    serialization_cursor_ptr c = make_cursor(os);
+    serialize_begin(c);
+    while (!c->segments_done) {
+        serialize_segments(c);
+
+        co_await write_iobuf_to_output_stream(
+          serialized.share(0, serialized.size_bytes()), output);
+        serialized.clear();
+
+        if (iso != _insync_offset) {
+            co_await output.close();
+            throw std::runtime_error(fmt_with_ctx(
+              fmt::format,
+              "Manifest changed duing serialization, in sync offset moved from "
+              "{} to {}",
+              iso,
+              _insync_offset));
+        }
+    }
```

Comment:
Is this something that will routinely happen?

Reply:
The insync offset advances with every batch processed by the STM, so if a batch is applied while the manifest is being serialized we'll throw. The alternative here is to lock the manifest while serializing it for the request, but I don't like user input being able to lock up the STM. I suppose I could add some retry logic at the request level, but my idea was for the user to retry this until successful.

Reply:
Hm. This is actually problematic since the response is chunked, which means that the caller can get a truncated manifest if the insync offset changes.

Reply:
As a user, i would be ok with a truncated response if it does not break the json parser

Reply:
I ended up taking the manifest lock, but placed an upper bound on the amount of time it can be held (3s).